cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterOperatorPKCS12AuthIdentity.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Represents the identity used by the cluster operator during TLS client authentication in the PKCS12 format.
+ * Can be used by clients that are unable to use the PEM format provided by PemAuthIdentity.
+ */
+public class ClusterOperatorPKCS12AuthIdentity {
+    private static final String SECRET_KEY = "cluster-operator";
+    private final byte[] keyStore;
+    private final String password;
+
+    /**
+     * Constructs the ClusterOperatorPKCS12AuthIdentity.
+     * @param secret Kubernetes Secret containing the client authentication identity
+     */
+    public ClusterOperatorPKCS12AuthIdentity(Secret secret) {
```

Comment:
What is the motivation to use a class instead of using a record with static method that creates it from the Secret?

Reply:
No motivation other than habit of creating classes and the other classes (PemTrustSet and PemAuthIdentity) don't work as records do they because the getter methods have some logic in them? I did originally have more logic in the constructors for them but as we discussed the other day that caused issues for the tests.

I am happy to change this to a record though if you want

Reply:
I can live with it if records do not work for the other classes you add for this.